s3repo/model.py

```diff
@@ -208,6 +181,177 @@ def _upload_files(self, package, tarantool_series, origin_files):
+                                    self._get_rpm_repo_path,
+                                    (path, dist_description['versions'])
+                                )
+                            )
+                        else:
+                            raise RuntimeError('Unknown repository base: ' +
+                                dist_description['base'])
+
+            # Collect the information about location of all the
+            # repositories from the bucket together.
+            for res in result_list:
+                repos_list.extend(res.get())
+
+        return repos_list
+
+    def get_supported_repos(self):
+        """Get description of the currently supported repos."""
+        return self.s3_settings['supported_repos']
+
+    def sync_all_repos(self):
+        """Update the metainformation of all known repositories."""
+        # Get the information about location of all the
+        # repositories from the bucket.
+        repos_to_update = self._get_repository_list()
+
+        # Add the repositories to the unsync list.
+        self.sync_lock.acquire()
+        for repo in repos_to_update:
+            self.unsync_repos.add(repo)
+        self.sync_lock.release()
+
+        # Add additional workers to update metainformation (approximate
+        # number of repositories to be synced ~ 600).
+        # 20 - the number up on the spot. Perhaps it will be corrected later.
+        threads_num = 20
+        with ThreadPool(processes=threads_num) as pool:
+            result_list = []
+            for _ in range(0, threads_num):
+                result_list.append(pool.apply_async(self.sync, (False,)))
+            for res in result_list:
+                # Wait for all additional workers to complete.
+                res.wait()
+
+    def sync(self, permanent):
+        """Update a metainformation of repositoties from the "unsync_repo" set.
+        permanent(bool) - describes whether the function should process data
+        permanent or whether it can "return" if all current work has been
+        completed.
+        """
+        while True:
+            if self.unsync_repos:
+                self.sync_lock.acquire()
+                sync_repo = self.unsync_repos.pop()
+                self.sync_lock.release()
+
+                mkrepo_cmd = [
+                    'mkrepo',
+                    '--s3-access-key-id',
+                    str(self.s3_settings['access_key_id']),
+                    '--s3-secret-access-key',
+                    str(self.s3_settings['secret_access_key']),
+                    '--s3-endpoint',
+                    str(self.s3_settings['endpoint_url']),
+                    '--s3-region',
+                    str(self.s3_settings['region']),
+                    's3://' + self.s3_settings['bucket_name'] + '/' + sync_repo
+                ]
+                with sp.Popen(mkrepo_cmd) as mkrepo_ps:
+                    result = mkrepo_ps.wait()
+                    if result != 0:
```

Comment:
`if result:` is more pythonic

Reply:
Yep. I write it so because `if result` seems like some kind of "positive" result, but in our case it's a "negative" case.